IThenticate.inc.php

```diff
@@ -0,0 +1,438 @@
+<?php
+
+/**
+ * @file IThenticate.inc.php
+ *
+ * Copyright (c) 2003-2024 Simon Fraser University
+ * Copyright (c) 2003-2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file LICENSE.
+ *
+ * @brief Service class to handle API communication with iThenticate service
+ */
+
+class IThenticate
+{
+    /** 
+     * The base api url in the format of schems://host
+     * 
+     * @var string
+     */
+    protected $apiUrl;
+
+    /**
+     * The required API Key to make the request
+     * @see https://developers.turnitin.com/docs/tca#required-headers
+     * 
+     * @var string
+     */
+    protected $apiKey;
+
+    /**
+     * Describes the platform/plugin integrating with TCA
+     * @see https://developers.turnitin.com/docs/tca#required-headers
+     * 
+     * @var string
+     */
+    protected $integrationName;
+
+    /**
+     * The version of the code that is integrating with TCA
+     * @see https://developers.turnitin.com/docs/tca#required-headers
+     * 
+     * @var string
+     */
+    protected $integrationVersion;
+    
+    /**
+     * The EULA(end user license agreement) that user need to confrim before making request
+     * 
+     * @var string|null
+     */
+    protected $eulaVersion = null;
+
+    /**
+     * The EULA details for a specific version
+     * 
+     * @var array|null
+     */
+    protected $eulaVersionDetails = null;
+
+    /**
+     * Base API path
+     * The string `API_URL` need to be replaced with provided api url to generarte fully qualified base url
+     * 
+     * @var string
+     */
+    protected $apiBasePath = "API_URL/api/v1/";
```

Comment:
Are we not using the v2 API?

Reply:
As per the confirmation , this `v1` is the latest of new API upgrade .